tokio/src/task/yield_now.rs

```diff
@@ -3,7 +3,17 @@ use std::pin::Pin;
 use std::task::{Context, Poll};
 
 doc_rt_core! {
-    /// Yield execution back to the Tokio runtime.
+    /// Return a `Future` that can be `await`-ed to yield execution back to the
+    /// Tokio runtime.
+    ///
+    /// A task yields by awaiting the returned `Future`, and may resume when
+    /// that future completes (with no output.) The current task will be
+    /// re-added as a pending task at the _back_ of the pending queue. Any
+    /// other pending tasks will be scheduled. No other waking is required for
+    /// the task to continue.
+    ///
+    /// See also the usage example in the [task module](index.html#yield_now).
+    #[must_use = "yield_now does nothing unless polled/`await`-ed"]
```

Comment:
Hmm, I feel like there are probably other async fns in tokio that don't return a value and also need a `must_use`, if they don't have one already!

Reply:
If you'd like I can go hunt for those... but I'd rather do it after this is merged, separate PR?